go/l1client/ethereum_client.go

```diff
@@ -0,0 +1,209 @@
+package l1client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/obscuronet/obscuro-playground/go/l1client/rollupcontractlib"
+	"github.com/obscuronet/obscuro-playground/go/l1client/wallet"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var connectionTimeout = 15 * time.Second
+
+type ethNode struct {
+	client          *ethclient.Client           // the underlying eth rpc client
+	id              common.Address              // TODO remove the id common.Address
+	wallet          wallet.Wallet               // wallet containing thea ccount information // TODO this does not need to be coupled together
+	chainID         int                         // chainID is used to sign transactions
+	txHandler       rollupcontractlib.TxHandler // converts L1TxData to ethereum transactions
+	contractAddress common.Address              // rollup contract address
```

Comment:
@otherview This field is never used. Does it need to be held per node, anyway? Isn't it the same across all nodes?

Reply:
It has to be used, but to me it feels like it should belong in the TxHandler thing ( or whatever we'll call it)

Reply:
The contract Address is required by the TxHandler. But I removed it from the struct 👍 
We can add it later if needed.